src/node/NodeHttpConfigFetcher.ts

```diff
@@ -72,30 +88,37 @@ export class NodeHttpConfigFetcher implements IConfigFetcher {
               },
             });
           } catch (err) {
-            options.logger.log(LogLevel.Error, 0, FormattableLogMessage.from("PROXY")`Failed to parse \`options.proxy\`: '${this.proxy}'.`, err);
+            this.logger?.log(LogLevel.Error, 0, FormattableLogMessage.from("PROXY")`Failed to parse \`options.proxy\`: '${this.proxy}'.`, err);
           }
         }
 
-        const headers: http.OutgoingHttpHeaders = {
-          "User-Agent": options.clientVersion,
-        };
-        if (lastEtag) {
-          headers["If-None-Match"] = lastEtag;
-        }
+        const { lastETag, timeoutMs } = request;
 
-        const requestOptions: http.RequestOptions | https.RequestOptions = {
+        const requestOptions: (http.RequestOptions | https.RequestOptions) & { headers?: Record<string, http.OutgoingHttpHeader> } = {
           agent,
-          headers,
-          timeout: options.requestTimeoutMs,
+          timeout: timeoutMs,
         };
-        options.logger.debug(JSON.stringify(requestOptions));
 
-        const request = (isBaseUrlSecure ? https : http).get(baseUrl, requestOptions, response => this.handleResponse(response, resolve, reject))
+        if (isCustomUrl) {
```

Comment:
In "backend" SDKs we can send any headers, it will not mess with the CORS preflight requests.